management/vmutils/vmutils.go

```diff
@@ -99,3 +99,48 @@ func ConfigureWindowsToJoinDomain(role *vm.Role, username, password, domainToJoi
 
 	return nil
 }
+
+func ConfigureWinRMListener(role *vm.Role, protocol vm.WinRMProtocol, certificateThumbprint string) error {
+
+	if role == nil {
+		return fmt.Errorf(errParamNotSpecified, "role")
+	}
+
+	winconfig := findConfig(role.ConfigurationSets, vm.ConfigurationSetTypeWindowsProvisioning)
+
+	if winconfig != nil {
+
+		listener := vm.WinRMListener{
+			Protocol:              protocol,
+			CertificateThumbprint: certificateThumbprint,
+		}
+
+		if winconfig.WinRMListeners == nil {
+			winconfig.WinRMListeners = &[]vm.WinRMListener{}
+		}
+
+		currentListeners := *winconfig.WinRMListeners
+
+		// replace existing listener if it's already configured
+		for i, existingListener := range currentListeners {
+			if existingListener.Protocol == protocol {
+				currentListeners[i] = listener
+				return nil
+			}
+		}
+
+		// otherwise append to list of listeners
+		newListeners := append(currentListeners, listener)
+		winconfig.WinRMListeners = &newListeners
+	}
```

Comment:
Shouldn't there be an error if this is not a windows config? Silent fail is maybe not the way to go... ;-)